app/components/rails_bootstrap/badge_component.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class RailsBootstrap::BadgeComponent < ApplicationComponent
+  
+  VARIRANTIONS = ["primary", "secondary", "success", "danger", "warning", "light", "info", "dark"].freeze
+
+  validates :variantion, inclusion: { in: VARIRANTIONS }
+  validates :pill, inclusion: { in: [true, false] }
+
+  def initialize(variantion: "primary", pill: false)
+    @variantion = "badge-#{variantion}"
```

Comment:
```suggestion
    @variation= "badge-#{variantion}"
```

Reply:
I will also prefer having `@variation` initialized to `variation`  paramter. If you think about it, its not the responsibility of `initilaizer`  class to handle this. We can make a new method and do something like this: 

```ruby
 def styles
   pill : "badge-#{variation}" :  "badge-pill"
 end
```

Reply:
You can then use this in the template file as : 

```erb
<span class"badge <%= styles %>"> <%= content %></span>
```

Reply:
```badge-variation``` and ```badge-pill``` are separated style. I will do something like that by adding another method.

Reply:
@amanpatel123 add styles function. need help to check the format of the codes.